src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php

```diff
@@ -19,18 +19,27 @@
  *
  * @author Christophe Coevoet <stof@notk.org>
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Diego Saint Esteben <diego@saintesteben.me>
  */
 class DecoratorServicePass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
+        $definitions = new \SplPriorityQueue();
```

Comment:
@stof Thanks for your feedback, I missed up that part of the documentation. Using arrays and `krsort` is the way to go?